### ISSUE TITLE
[BUG]: is this an issue with deepface running as a container?

### ISSUE THREAD

**Issue by levski** [serengil/deepface#1419]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [X] I have updated to the latest version of the packages.
- [X] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.94

### Python version

3.8.12

### Operating System

debian 12

### Dependencies

root@1cfe87051553:/app/deepface/api/src# pip freeze
absl-py==2.1.0
astunparse==1.6.3
beautifulsoup4==4.12.3
blinker==1.8.2
cachetools==5.5.0
certifi==2024.12.14
charset-normalizer==3.4.1
click==8.1.8
# Editable install with no version control (deepface==0.0.94)
-e /app
filelock==3.16.1
fire==0.7.0
Flask==3.0.3
Flask-Cors==5.0.0
flatbuffers==24.12.23
gast==0.4.0
gdown==5.2.0
google-auth==2.37.0
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
grpcio==1.68.1
gunicorn==23.0.0
h5py==3.11.0
idna==3.10
importlib_metadata==8.5.0
itsdangerous==2.2.0
Jinja2==3.1.5
keras==2.13.1
libclang==18.1.1
Markdown==3.7
MarkupSafe==2.1.5
mtcnn==0.1.1
numpy==1.22.3
oauthlib==3.2.2
opencv-python==4.9.0.80
opt_einsum==3.4.0
packaging==24.2
pandas==2.0.3
Pillow==9.0.0
protobuf==4.25.5
pyasn1==0.6.1
pyasn1_modules==0.4.1
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2024.2
requests==2.32.3
requests-oauthlib==2.0.0
retina-face==0.0.17
rsa==4.9
six==1.17.0
soupsieve==2.6
tensorboard==2.13.0
tensorboard-data-server==0.7.2
tensorflow==2.13.1
tensorflow-estimator==2.13.0
tensorflow-io-gcs-filesystem==0.34.0
termcolor==2.4.0
tqdm==4.67.1
typing_extensions==4.5.0
tzdata==2024.2
urllib3==2.2.3
Werkzeug==3.0.6
wrapt==1.17.0
zipp==3.20.2
root@1cfe87051553:/app/deepface/api/src# 

### Reproducible example

```Python
# Base image
FROM python:3.8.12
LABEL org.opencontainers.image.source https://github.com/serengil/deepface

# Create required folder
RUN mkdir -p /app && chown -R 1001:0 /app
RUN mkdir /app/deepface

# Switch to application directory
WORKDIR /app

# Install system dependencies
RUN apt-get update && apt-get install -y \
    wget \
    ffmpeg \
    libsm6 \
    libxext6 \
    libhdf5-dev \
    && rm -rf /var/lib/apt/lists/*

# Add NVIDIA repositories and install CUDA toolkit and cuDNN
RUN wget https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/cuda-keyring_1.1-1_all.deb && \
    dpkg -i cuda-keyring_1.1-1_all.deb && \
    apt-get update && apt-get install -y \
    cuda-toolkit-12-6 \
    cudnn \
    cudnn-cuda-12 \
    && rm -rf /var/lib/apt/lists/*

# Copy application files
COPY ./deepface /app/deepface
COPY ./requirements.txt /app/requirements.txt
COPY ./requirements_local /app/requirements_local.txt
COPY ./package_info.json /app/
COPY ./setup.py /app/
COPY ./README.md /app/
COPY ./entrypoint.sh /app/deepface/api/src/entrypoint.sh

# Upgrade pip
RUN pip install --no-cache-dir --upgrade pip

# Install TensorFlow with GPU support
RUN pip install --no-cache-dir tensorflow[and-cuda]

# Install DeepFace dependencies
RUN pip install --no-cache-dir -r /app/requirements_local.txt
RUN pip install --no-cache-dir -e .

# Environment variables
ENV PYTHONUNBUFFERED=1

# Configure app directory and port
WORKDIR /app/deepface/api/src
EXPOSE 5100
ENTRYPOINT [ "sh", "entrypoint.sh" ]
```

### Relevant Log Output

junjun@i-c-u:/opt/deepface$ docker exec -it deepface-gpu bash
python -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
root@1cfe87051553:/app/deepface/api/src# python -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
2025-01-04 17:28:39.830945: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2025-01-04 17:28:39.883713: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2025-01-04 17:28:39.884144: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-01-04 17:28:41.048806: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2025-01-04 17:28:42.174258: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
2025-01-04 17:28:42.227964: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1960] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
[]

### Expected Result

gpu detected and used

### What happened instead?

container will use the cpu

### Additional Info

other containers happily using the gpu.
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 560.35.05              Driver Version: 560.35.05      CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce GTX 1080 Ti     Off |   00000000:0E:00.0 Off |                  N/A |
| 45%   63C    P2             82W /  280W |    7805MiB /  11264MiB |     23%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      9727      C   ...in/ubuntu/python38/venv/bin/python3       1334MiB |
|    0   N/A  N/A     10034      C   ...in/ubuntu/python38/venv/bin/python3        728MiB |
|    0   N/A  N/A     48866      C   ...in/ubuntu/python38/venv/bin/python3        890MiB |
|    0   N/A  N/A   1349422      C   /usr/bin/python                               830MiB |
|    0   N/A  N/A   1710368      C   ffmpeg                                        424MiB |
|    0   N/A  N/A   1710373      C   ffmpeg                                        424MiB |
|    0   N/A  N/A   1710381      C   ffmpeg                                        528MiB |
|    0   N/A  N/A   1710384      C   ffmpeg                                        528MiB |
|    0   N/A  N/A   2037053      C   ffmpeg                                        418MiB |
|    0   N/A  N/A   2037056      C   ffmpeg                                        424MiB |
|    0   N/A  N/A   2037058      C   ffmpeg                                        424MiB |
|    0   N/A  N/A   2037243      C   ffmpeg                                        424MiB |
|    0   N/A  N/A   2037246      C   ffmpeg                                        424MiB |
+-----------------------------------------------------------------------------------------+

is this issue present in baremetal installs or lxd?
just trying to understand if this is an issue with deepface running in a container.

**Comment by serengil** [comment]
answered this many times, deepface does nothing with gpu

deepface depends tensorflow and tensorflow can use gpu if your environment is okay

**Comment by jnxyatmjx** [comment]
> answered this many times, deepface does nothing with gpu
> 
> deepface depends tensorflow and tensorflow can use gpu if your environment is okay

can deepface imporve perfromance if use tensorflow with gpu？I use jetson tx2 platforme
